### PR TITLE
Cleanup about cutoffs.

### LINKED ISSUE #423: Cleanup the regulations about cutoffs
I just did a careful read of how the regulations talk about cutoffs (with the intent of documenting these things in our competition JSON format (see https://github.com/thewca/worldcubeassociation.org/issues/946)). There are some ambiguities and inconsistencies with how the relevant regulations are worded that I think we could clean up.

The regulations define 2 kinds of cutoffs:

- [9g2](https://www.worldcubeassociation.org/regulations/#9g2): Advancing to the next phase of a combined round ("decided by ranking (best x competitors) or by result").
- [9p2](https://www.worldcubeassociation.org/regulations/#9p2): Advancing to the next round of an event ("determined by either a cutoff ranking or a cutoff time in the preceding elimination round").

## My complaints

1. The regulations do not define good terms to distinguish between these 2 concepts. As things stand right now, it's ambiguous what "cutoff" refers to. IMO, this ambiguity is part of why it has been so difficult to stamp out usage of the phrases "hard cutoff" and "soft cutoff".
2. 9p2 uses the phrase "cutoff time", which seems to exclude 333mbf and 333fm, events which are not ranked by time (see #429).
3. 9p2 uses the phrase "elimination round", which shows up nowhere else in the regulations. I know what it's referring to, but I think the word "elimination" only serves to increase cognitive load, forcing the reader to ask: "how does an elimination round differ from a regular round?"

## How I'd address my complaints

1. Invent new terms to distinguish between "next phase cutoff" and "next round cutoff" and remove all references to vanilla "cutoff" from the regulations. Here are some suggestions:
    - 9g2: "next phase cutoff", "phase cutoff", "combined round cutoff", "combined cutoff", or "soft cutoff"
    - 9p2: "next round cutoff", "round cutoff", "advancing cutoff"
2. Change the phrasing of 9p2 to be exactly the same as the phrasing of 9g2: "decided by ranking (best x competitors) or by result".
3. Remove the word "elimination", I don't believe it adds any value here.

I'm happy to send in a PR with some or all of these changes if you like these ideas.

### LINKED ISSUE #426: Clarification on how exactly 9g2 cutoffs by result work
Emphasis mine:

> [9g2](https://www.worldcubeassociation.org/regulations/#9g2)) Whether a competitor proceeds to next phase of a Combined round, must be decided by ranking (best x competitors) or **by result (all competitors with a best result under x)** of the first phase.

Just to be clear about this, we have 3 "kinds" of results, depending on the event:

- FMC: Integer (number of turns) or DNF or DNS
- Multi bld: 2 integers + centiseconds or DNF or DNS
- Every other event: Centiseconds or DNF or DNS

A liberal interpretation of regulation 9g2 makes it sound like you could theoretically have a combined round of FMC mean of 3 where people must solve the first attempt in less than or equal to 30 moves in order to be allowed to participate in the second and third attempts.

Things get weird for multibld. [9f12c](https://www.worldcubeassociation.org/regulations/#9f12c) defines what a "result" is for mbld, and it says that "a greater difference [result] is better". This, combined with the fact that 9g2 says "all competitors with a best result **under** x" results in a funny conclusion: you'd have to do "worse" to advance to the next phase a combined round of mbld. (aplogies if I got this wrong, I read the relevant regulations *many* times before writing this).

Is there any reason to ever hold a combined round of FMC or a combined round of multibld? Supporting this will make our software significantly more complicated than I feel it is worth doing. Similarly with #425, in practice I bet these never happen, in part because cubecomps just doesn't support it.

I propose extending 9g2 to only allow combined rounds for events where results are measured in centiseconds.

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
Fixes #423 and partly #429.

### CODE REVIEW

**Comment by AlbertoPdRF** [comment]
With the latest commit, both #423 and #429 are addressed. Let's discuss if there are better solutions :)

EDIT: I wrote 426 instead of 423 in the first commit, so I'll address that issue in this PR too. After all, those 3 issues are related one with another :)

File: wca-regulations.md [hunk 1]
```diff
@@ -248,7 +248,7 @@ Note: Because Article and Regulation numbers are not reassigned when Regulations
     - 9f14) For "Mean of 3" and "Average of 5" rounds, if two or more competitors achieve identical average/mean results, rankings are assessed based on the best attempt per competitor, where "better" is defined as the shorter time.
     - 9f15) Competitors who achieve the same result in a round receive an identical ranking for the round.
 - 9g) A Combined Round consists of two phases of attempts, where competitors advance to the second phase if they meet a designated cutoff during the first phase.
-    - 9g2) Whether a competitor proceeds to next phase of a Combined round, must be decided by ranking (best x competitors) or by result (all competitors with a best result under x) of the first phase.
+    - 9g2) Whether a competitor proceeds to next phase of a Combined round, must be decided by ranking (best x competitors) or by result (all competitors with a better result than x) of the first phase.
```
Comment (jfly) [inline wca-regulations.md:251]: Might it make sense to link to [9f12](https://www.worldcubeassociation.org/regulations/#9f12) here after the use of the word "better"?

**Review by jfly — COMMENTED** [review]
This looks great to me!

File: wca-regulations.md (hunk 1 above)
Comment (AlbertoPdRF) [reply wca-regulations.md:251]: @thewca/wrc-team Thoughts? I personally think it's a good idea!

**Comment by Laura-O** [comment]
LGTM!

File: wca-regulations.md (hunk 1 above)
Comment (jfly) [reply wca-regulations.md:251]: Bump, this looks like it got merged without this chance happening. Thoughts?

Comment (Laura-O) [reply wca-regulations.md:251]: Yeah, I completely missed that before merging. Thanks for checking, @jfly

Comment (AlbertoPdRF) [reply wca-regulations.md:251]: Addressed on #478

**Comment by pedrosino** [comment]
I believe there's an error here:
Regulation
> 9p2) The competitors who advance to the next round must be determined by ranking (best x competitors) or by result (all competitors with a better result than x) in the preceding round.

Guideline
> 9p2+) [CLARIFICATION] Best result refers to the single or average that determines the ranking of the corresponding round.

There's no "best result" in the regulation anymore (it used to be, but at 9g2), so the guideline doesn't make sense. Also, "better result than" sounds weird. I talked to @jfly and he agrees that "result better than" is nicer.

Actually, there's no need to talk about a "best result" in either place. You need A result better than X.

Here's a suggestion:
> 9g2) Whether a competitor proceeds to next phase of a Combined round must be decided by result (all competitors with a **result better than** x) of the first phase (see [Regulation 9f12](regulations:regulation:9f12)).

> 9p2) The competitors who advance to the next round must be determined by ranking (best x competitors) or by result (all competitors with a **result better than** x) in the preceding round.

> 9p2+) Result refers to the single or average that determines the ranking of the corresponding round.


EDIT

Comparison of current text and my suggestion:
https://www.diffchecker.com/v1CdIXKv

**Pull request merged**